omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskDefinitionController1_8Test.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_8;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.MockTaskServiceWrapper;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskDefinitionResource1_8;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+import java.util.List;
+
+public class TaskDefinitionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition testTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		testTask.setRepeatInterval(10L);
+		testTask.setStartOnStartup(true);
+		testTask.setProperty("propertyKey", "propertyValue");
+		
+		mockTaskServiceWrapper.registeredTasks.add(testTask);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	@Override
+	public void shouldGetAll() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		SimpleObject resultTasks = deserialize(handle(req));
+		List<Object> results = Util.getResultsList(resultTasks);
+		Assert.assertNotNull(resultTasks);
+		Assert.assertEquals(results.size(), getAllScheduledCount());
+	}
+	
+	@Test
+	public void shouldGetAllRegistered() throws Exception {
```

Comment:
What is the difference between shouldGetAll and shouldGetAllRegistered?

Reply:
Defaultly GetAll method in the resource has been set to return the list of Scheduled tasks. (I can't change the name because it is an Overriden method)
So I am using another separate method to test about the Registered tasks.

Reply:
Isn't it more natural for GetAll to return all tasks regardless of whether they are scheduled or not?

Reply:
So I will change the way to retrieve all the registered for the getAll method